src/views/SemanticWorkflow/Definitions/Sheet.lua

```diff
@@ -0,0 +1,49 @@
+---@diagnostic disable:missing-return
+
+---@class Sheet
+---@field public preview_frame SelectionFrame The frame to which to proceed when re-running the game after a change.
+---@field public active_frame SelectionFrame The frame whose controls to display in the "Inputs" page.
+---@field public sections Section[] The sections to execute in order.
+---@field public name string A name for the sheet for convenience.
+---@field public busy boolean Whether the sheet is waiting for the game to run until its preview frame.
+---@field private _section_index integer The nth section that is currently being played.
+---@field private _frame_counter integer The nth frame of the current section that is currently being played.
+---@field private _update_pending boolean Whether a change has been made that demands rerunning the sheet until its preview frame.
+---@field private _savestate unknown The savestate this sheet runs from.
```

Comment:
Savestates have a type. 

Consider including the [Mupen64 Lua API definitions](https://github.com/mupen64/mupen64-rr-lua/blob/master/src/api.lua) in the `lib/` directory to reference the savestate type (`ByteBuffer`)